### PR TITLE
dispatch: use Snappy S2 compression and vtprotobuf

### PR DESCRIPTION
**Pull request by jzelinskie**

### CODE REVIEW

File: pkg/proto/dispatch/v1/dispatch_vtproto.pb.go [hunk 1]
```diff
@@ -0,0 +1,4497 @@
...
+type UnsafeDispatchServiceServer interface {
+	mustEmbedUnimplementedDispatchServiceServer()
+}
+
+func RegisterDispatchServiceServer(s grpc.ServiceRegistrar, srv DispatchServiceServer) {
+	s.RegisterService(&DispatchService_ServiceDesc, srv)
+}
+
+func _DispatchService_DispatchCheck_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+	in := DispatchCheckRequestFromVTPool()
```
Comment (jzelinskie) [inline pkg/proto/dispatch/v1/dispatch_vtproto.pb.go:188]: vtprotobuf replaces our usage of the grpc plugin and generates these handlers that grab messages from the pool.

File: pkg/proto/dispatch/v1/01_codec.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 The Vitess Authors.
```
Comment (jzelinskie) [inline pkg/proto/dispatch/v1/01_codec.go:1]: This codec tells gRPC that it should try to use vtproto first and fallback to regular proto parsing.

File: buf.gen.yaml [hunk 3]
```diff
@@ -13,9 +13,9 @@ plugins:
   - name: "go"
     out: "."
     opt: "paths=source_relative"
-  - name: "go-grpc"
+  - name: "go-vtproto"
     out: "."
-    opt: "paths=source_relative"
+    opt: "paths=source_relative,features=marshal+unmarshal+size+clone+pool+grpc,pool=github.com/authzed/spicedb/pkg/proto/dispatch/v1.DispatchCheckRequest,pool=github.com/authzed/spicedb/pkg/proto/dispatch/v1.DispatchExpandRequest,pool=github.com/authzed/spicedb/pkg/proto/dispatch/v1.DispatchLookupRequest,pool=github.com/authzed/spicedb/pkg/proto/dispatch/v1.DispatchReachableResourcesRequest,pool=github.com/authzed/spicedb/pkg/proto/dispatch/v1.DispatchLookupSubjects"
```
Comment (josephschorr) [inline buf.gen.yaml:18]: Can we add a comment about how this was created?

File: pkg/proto/dispatch/v1/01_codec.go (hunk 2 above)
Comment (josephschorr) [reply pkg/proto/dispatch/v1/01_codec.go:1]: Can we add that as a comment somewhere in here?

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**